src/tools/user-agent-explainer/user-agent-explainer.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <div>
+    <n-space justify="center">
+      <n-form-item label="Use user agent from local browser?" label-placement="left" :show-feedback="false">
+        <n-switch v-model:value="useCurrentUserAgent" @click="useCurrentUserAgentClick" />
+      </n-form-item>
+    </n-space>
+    <n-form-item label="User agent string" v-bind="validationAttrs">
+      <n-input v-model:value="ua" :disabled="useCurrentUserAgent" />
+    </n-form-item>
+
+    <user-agent-result-cards :user-agent-info="userAgentInfo" :sections="sections" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue';
+import { useStorage } from '@vueuse/core';
+import { UAParser } from 'ua-parser-js';
+import { withDefaultOnError } from '@/utils/defaults';
+import { useValidation } from '@/composable/validation';
+import { isNotThrowing } from '@/utils/boolean';
+import { Adjustments, Browser, Cpu, Devices, Engine } from '@vicons/tabler';
+import UserAgentResultCards from './user-agent-result-cards.vue';
+
+const ua = useStorage('user-agent-explainer:ua', navigator.userAgent);
+const useCurrentUserAgent = useStorage('user-agent-explainer.useCurrentUserAgent', true);
+
+const getUserAgentInfo = (userAgent: string) => new UAParser(userAgent.trim());
```

Comment:
Maybe prevent the call to every UAParser method by calling getResults()
```suggestion
const getUserAgentInfo = (userAgent: string) => new UAParser(userAgent.trim()).getResults();
```

Reply:
Fixed, but use the shorthand version `UAParser(userAgent.trim())` instead of `new UAParser(userAgent.trim()).getResults()`.